### PR TITLE
A few fixes for the feature form date widget, the layer tree pop up, and file picker

### PR DESCRIPTION
**Pull request by nirvn**
First PR, let's see if that'll pass tests.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-e868c985e2a5a0a98ab970399cc2f5e633ea9854-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-e868c985e2a5a0a98ab970399cc2f5e633ea9854-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-e868c985e2a5a0a98ab970399cc2f5e633ea9854-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-e868c985e2a5a0a98ab970399cc2f5e633ea9854-x86-5.apk))

File: src/qml/editorwidgets/DateTime.qml [hunk 1]
```diff
@@ -37,20 +37,23 @@ Item {
...
 
       TextField {
         id: label
 
-        anchors.fill: parent
+        anchors.left: parent.left
+        anchors.right: parent.right
         verticalAlignment: Text.AlignVCenter
-        font.pointSize: 16
+        font.pointSize: 18 * dp
```
Comment (m-kuhn) [inline src/qml/editorwidgets/DateTime.qml]: Are you sure the **point**Size needs to be multiplied with dp? I think it's only the **pixel**Size and the pointSize is fine, just like that.

File: src/qml/ui/LayerTreeItemProperties.ui.qml [hunk 2]
```diff
@@ -5,16 +5,22 @@ import QtQuick.Layouts 1.3
...
-            font.pointSize: 14
+            font.bold: true
+            font.pointSize: 18 * dp
         }
 
         CheckBox {
             id: itemVisibleCheckBox
             text: qsTr("Show on map canvas")
-            font.pointSize: 14
+            font.pointSize: 18 * dp
```
Comment (nirvn) [inline src/qml/ui/LayerTreeItemProperties.ui.qml]: @m-kuhn , re pointSize, this checkbox had a pointSize of 14 (without the dp multiplier) and looks super super tiny on my smartphone.

**Comment by Emic37** [comment]
Screenshot_20190926-231720_QField Beta 627
Oh huge! I think there is an error... 😁

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-70e13c516ad7ea3f4588f8be6775dd62040c8fc6-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-70e13c516ad7ea3f4588f8be6775dd62040c8fc6-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-70e13c516ad7ea3f4588f8be6775dd62040c8fc6-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-70e13c516ad7ea3f4588f8be6775dd62040c8fc6-x86-5.apk))

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-2096954dd2e3ba5456a25633c5621ef491101c1b-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-2096954dd2e3ba5456a25633c5621ef491101c1b-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-2096954dd2e3ba5456a25633c5621ef491101c1b-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-2096954dd2e3ba5456a25633c5621ef491101c1b-x86-5.apk))

**Comment by signedav** [comment]
Thanks for the implementations @nirvn . The part with the layer popup looks fine :+1: 

The date widget seems IMHO not harmonized with this implementation #646 (see below)

### Colors 
Since there for comboboxes after #646 the background is gray on editing and transparent on read only. I would have it here the same. Means gray on editing like before, and transparent in read only.

### Underscore
It looks a little noisy on the device with the frame and the underscore.
image

I would remove the underscore and put the text into the vertical center (like the combobox in #646)

**Comment by signedav** [comment]
Another thing I want to mention but this is **not related to this** implementation:

While the selection of the date debugging QField on the computer looks fine like this...
datetime-debug

... it's not really comfortable on the device, since there is no indication, that I pressed a number (the small circle is only the cursor I activated on the touch device)... 
datetime

**Comment by nirvn** [comment]
@signedav , comments addressed.

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-098f314c5b8d9c6d8c699fc54da91a74c463a00a-armv7-X.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-098f314c5b8d9c6d8c699fc54da91a74c463a00a-x86-X.apk) (Android 5: [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-098f314c5b8d9c6d8c699fc54da91a74c463a00a-armv7-5.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-098f314c5b8d9c6d8c699fc54da91a74c463a00a-x86-5.apk))

**Comment by signedav** [comment]
Thanks @nirvn looks better now.

Only thing left (it has been like that before your PR as well) that the label is not centered though of the `verticalAlignment: Text.AlignVCenter` 

image

Maybe you can fix it with small effort...

File: src/qml/editorwidgets/DateTime.qml [hunk 3]
```diff
@@ -37,20 +37,23 @@ Item {
       Layout.minimumHeight: 48 * dp
 
       Rectangle {
-        anchors.fill: parent
         id: backgroundRect
+        anchors.fill: parent
```
Comment (signedav) [inline src/qml/editorwidgets/DateTime.qml]: Apparently `anchors.fill: label` does not work but:
```suggestion
        height: label.height - label.bottomPadding / 2
        width: label.width
```

Comment (nirvn) [reply src/qml/editorwidgets/DateTime.qml]: That doesn't work, rect's height becomes zero.

**Comment by nirvn** [comment]
@signedav , tried an alternative approach.

As far as I'm concerned, this can be merged.

File: src/qml/editorwidgets/DateTime.qml (hunk 3 above)
Comment (signedav) [reply src/qml/editorwidgets/DateTime.qml]: Okay, strange it worked on my system.
With your new approach, it set's the anchors. Beside that it does the same like i suggested, no?

Comment (nirvn) [reply src/qml/editorwidgets/DateTime.qml]: Yep, it's seemingly identical, yet on my system it failed to show anything using the alternative suggested. QML can work in mysterious ways ;)

Comment (signedav) [reply src/qml/editorwidgets/DateTime.qml]: > QML can work in mysterious ways

Yes. I experienced that :-)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-813828b80a617cfdf1e504fbe6bf5c670d77811d-armv7-X.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-813828b80a617cfdf1e504fbe6bf5c670d77811d-arm64_v8a-X.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-813828b80a617cfdf1e504fbe6bf5c670d77811d-x86-X.apk)
 - [armv7 (legacy build for Android 5)](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-813828b80a617cfdf1e504fbe6bf5c670d77811d-armv7-5.apk)
 - [x86 (legacy build for Android 5)](https://download.opengis.ch/qfield/ci-builds/qfield-dev-627-813828b80a617cfdf1e504fbe6bf5c670d77811d-x86-5.apk)

**Comment by nirvn** [comment]
@signedav , travis green 'n happy. Can we merge this PR?

**Comment by nirvn** [comment]
@signedav , ok, I'll investigate some more.

**Pull request merged**